openface_ros/scripts/face_recognition_node

```diff
@@ -4,40 +4,62 @@ import rospy
 from cv_bridge import CvBridge, CvBridgeError
 
 from image_recognition_msgs.srv import Recognize, Annotate
-from image_recognition_msgs.msg import Recognition, CategoryProbability, CategoricalDistribution
-from sensor_msgs.msg import RegionOfInterest
+from image_recognition_msgs.msg import Recognition, Recognitions, CategoryProbability, CategoricalDistribution
+from sensor_msgs.msg import RegionOfInterest, Image
 from std_srvs.srv import Empty
 
-import cv2
 import os
-from datetime import datetime
 import sys
 
 from image_recognition_util import image_writer
 from openface_ros.face_recognizer import FaceRecognizer, RecognizedFace
 
 
 class OpenfaceROS:
-    def __init__(self, align_path, net_path, save_images_folder):
+    def __init__(self, align_path, net_path, save_images_folder, topic_save_images,
+                 service_save_images, topic_publish_result_image, service_publish_result_image):
         """
         OpenfaceROS class that wraps the FaceRecognizer in a ROS node
         :param align_path: dlib face align path
         :param net_path: openface neural net path
         :param save_images_folder: path where to store the images
+        :param topic_save_images: whether to save images originated from image topic callback
```

Comment:
Nitpicking: I'd like to see the type of each param, e.g, I think ```bool``` in these cases

Reply:
Description is clear enough I.M.O.